grade.go

```diff
@@ -392,39 +404,27 @@ func main() {
 
 		result.StudentId = hydratedStudent.Id
 		result.AssignmentId = int8(1)
-
-		if language == "python3" || language == "python" {
+		
+		if language == "python3" || language == "python" || language == "javascript" || language == "js" {
 			result.CompileSuccess = true
-
-			stdout := runInterpreted(filepath.Join(workDir, dir), runArgs, input)
+	
+			stdout := runInterpreted(filepath.Join(workDir, dir), runArgs, language, input)
 			fmt.Printf("Output for %s: %s", result.Student, stdout)
 			result.RunCorrect, result.Feedback = compare(expected, stdout)
+			
+		// if student doesn't exist, commit to db
+		} else if language == "java" || language == "c++" {
+				gradeSubmission(&result, dir, workDir, runArgs, expected, language, input, wall)
 		} else {
-			result.CompileSuccess = compile(filepath.Join(workDir, dir), wall)
-			if result.CompileSuccess {
-				stdout := runCompiled(filepath.Join(workDir, dir), runArgs, input)
-				result.RunCorrect, result.Feedback = processOutput(expected, stdout)
-			}
+		fmt.Print("No language found")
 		}
-	}
 
+	}
 	for _, id := range results.Order {
-		fmt.Printf("%s: [compileSuccess=%t] [runCorrect=%t]\n", id, results.Results[id].CompileSuccess, results.Results[id].RunCorrect)
+		fmt.Printf("%s: [compileSuccess=%t] [runCorrect=%t] \n", id, results.Results[id].CompileSuccess, results.Results[id].RunCorrect)
 	}
-
 	if !isDryRun {
 		writeFullOutputToDb(supabase, results)
 	}
-
 	createCsv(results, outFile)
-
-	// This is for getting files in a directory, later to be searched with *.py, if that is how we end up implementing it
-	// files, err := ioutil.ReadDir(workDir)
-	// if err != nil {
-	// 	log.Fatal(err)
-	// }
-
-	//	for _, file := range files {
-	//		fmt.Println(file.Name(), file.IsDir())
-	//	}
-}
+}
```

Comment:
go fmt

Reply:
Oof, yes didn't know there would be that many of these. Going to remember to use go fmt from now on.

Reply:
if you're using vscode and have go tooling installed it _should_ use the go formatter by default (that's how i do it). You can enable format-on-save if you want as well.